src/pages/view/FilterBar/FilterBar.js

```diff
@@ -26,16 +57,147 @@ const FilterBar = ({ filters, removeFilter, removeAllFilters }) => {
 
     const closeDialog = () => setDialogIsOpen(false)
 
+    const closeFilterDialog = useCallback(() => {
+        setFilterDialogIsOpen(false)
+        setShowScope(true)
+    }, [activeFilter])
+    const onConfirmFilterDialog = useCallback(
+        async ({ name, visibility, id }) => {
+            await handleSaveFilter({ ...activeFilter, name, visibility, id })
+            closeFilterDialog()
+        },
+        [activeFilter, closeFilterDialog]
+    )
+
+    const hasActiveSavedFilter = Boolean(activeFilter.id)
+    const savedFilterHasChanges = useMemo(
+        () => hasActiveSavedFilter && !deepEqual(activeFilter.values, filters),
+        [activeFilter, filters]
+    )
+
+    const toggleMoreActions = () => {
+        setMoreOptionsIsOpen((prev) => !prev)
+    }
+    const showFilterAction = isFilterActionAllowed(activeFilter, currentUser)
+
+    const handleSaveFilter = useCallback(async (filter = {}) => {
+        await saveFilter(currentUser, {...activeFilter, ...filter})
+        setMoreOptionsIsOpen(false)
+    }, [currentUser, activeFilter])
+    const handleSaveNewFilter = useCallback(() => {
+        setFilterDialogData({ id: 'new' })
+        setFilterDialogIsOpen(true)
+    }, [activeFilter])
+    const handleRenameFilter = useCallback(() => {
+        setShowScope(false)
+        setFilterDialogData(activeFilter)
+        setFilterDialogIsOpen(true)
+        setMoreOptionsIsOpen(false)
+    }, [activeFilter])
+    const handleDeleteFilter = useCallback(async () => {
```

Comment:
Can we move all of these methods into a hook? Additionally, add spaces between methods to improve readability